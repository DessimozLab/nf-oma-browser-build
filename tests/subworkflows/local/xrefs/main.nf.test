nextflow_workflow {

    name "Test Workflow GENERATE_XREFS"
    script "subworkflows/local/xrefs/main.nf"
    workflow "GENERATE_XREFS"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                outdir = "tests/results"
                xref_refseq = null
                xref_uniprot_swissprot = "tests/data/xrefs/uniprot_swissprot_test.fasta"
                xref_uniprot_trembl = "tests/data/xrefs/uniprot_trembl_test.fasta"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.value([id: 'test_run', name: 'Test Run', description: 'A test run for unit testing', nr_of_sequences: 1000000])
                input[1] = Channel.fromPath('tests/data/xrefs/test_gs_xref.tsv')
                input[2] = Channel.fromPath('tests/data/oma_browser/test_db.h5')
                input[3] = Channel.fromPath('tests/data/oma_browser/test_seqidx.h5')
                input[4] = Channel.fromPath('tests/data/oma_browser/test_seqbuf.bin')
                input[5] = Channel.fromPath('tests/data/xrefs/test_source_xref_db.h5')
                """
            }
            // run in stub mode only!
            options "-stub-run"
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 20
            assert snapshot(workflow.out).match()
        }

    }

}
