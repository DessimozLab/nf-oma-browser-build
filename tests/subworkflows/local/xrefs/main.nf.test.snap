{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "taxmap.pkl:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "XRef-db.h5:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "reduced-xrefs-db.h5:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "red_xref_db": [
                    "reduced-xrefs-db.h5:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "taxmap": [
                    "taxmap.pkl:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "xref_db": [
                    "XRef-db.h5:md5,d41d8cd98f00b204e9800998ecf8427e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-10-29T09:00:44.685298"
    }
}